### ISSUE TITLE
Changing the type of a stack-based array does not take effect the first time

### ISSUE THREAD

**Issue by joelreymont** [Vector35/binaryninja-api#5414]
**Version and Platform (required):**
- Binary Ninja Version: 4.1.5272-dev, 3a057a87
- OS: macos
- OS Version: 14.4
- CPU Architecture: arm64

```
0006815c  struct ErrObj* DjiError_GetErrorMsgElements(int64_t err_code, struct ErrDesc* err_desc @ x8)

00068178      char* description
00068178      __builtin_memset(s: &description, c: 0, n: 0x20)
```

I pointed at `description` and set the type with 'y' to `char description[0x20]` this is what I got as a result

```
0006815c  struct ErrObj* DjiError_GetErrorMsgElements(int64_t err_code, struct ErrDesc* err_desc @ x8)
00068178      int64_t s
00068178      __builtin_memset(s: &s, c: 0, n: 0x20)
```

Typing `s` and changing it to `char description[0x20]` does the right thing 

```
0006815c  struct ErrObj* DjiError_GetErrorMsgElements(int64_t err_code, struct ErrDesc* err_desc @ x8)
00068178      char description[0x20]
00068178      __builtin_memset(s: &description, c: 0, n: 0x20)
```

but why not the first time around?

I was able to reproduce this several times by undoing my changes.

**Comment by negasora** [comment]
Here's a binary that repros the issue
1. Go to main
2. Change `var_38` to `char str[0x20]`
3. Change `str` to `char* str`
4. Notice that the user-defined name gets undefined and it's marked low-confidence
[a.txt](https://github.com/Vector35/binaryninja-api/files/15310624/a.txt)

**Comment by negasora** [comment]
Change it to `char* str` after

**Comment by joelreymont** [comment]
Yes, it went back to `s` and the type was greyed-out so must be low confidence.

My original complaint was that the process was going the other way around, though, and I had to ask BN to make my type twice. Did it decide it was a low-confidence type the first time around?

**Comment by negasora** [comment]
That's my current guess

**Comment by joelreymont** [comment]
I don't know if BN is working as intended here so feel free to close the issue.

I object to BN claiming that I'm wrong, though, given that I'm right in the end 🤷🏼‍♂️.

**Comment by negasora** [comment]
Definitely not working as intended :P
